libraries/stdlib/src/kotlin/util/Preconditions.kt

```diff
@@ -82,7 +89,8 @@ public inline fun <T:Any> checkNotNull(value: T?): T = checkNotNull(value) { "Re
  * @sample samples.misc.Preconditions.failCheckWithLazyMessage
  */
 @kotlin.internal.InlineOnly
-public inline fun <T:Any> checkNotNull(value: T?, lazyMessage: () -> Any): T {
+@kotlin.internal.Returns
+public inline fun <T:Any> checkNotNull(@kotlin.internal.Equals(kotlin.internal.ConstantValue.NOT_NULL) value: T?, lazyMessage: () -> Any): T {
```

Comment:
In the current implementation, we don't have a way to express that function always throws (mainly because we were focused on practical applications)